crates/sui-core/src/validator_tx_finalizer.rs

```diff
@@ -25,10 +28,16 @@ const TX_FINALIZATION_DELAY: Duration = Duration::from_secs(60);
 /// If a transaction can not be finalized within 1 min of being woken up, give up.
 const FINALIZATION_TIMEOUT: Duration = Duration::from_secs(60);
 
+/// Incremental delay for validators to wake up to finalize a transaction.
+const VALIDATOR_DELAY_INCREMENTS_SEC: u64 = 10;
```

Comment:
IIUC, the solution is to prevent byzantine fault. 10s seems a bit low here since that also falls into not-too-uncommon slowness. Therefore I wonder if this can be in minutes.